src/PimApi/PimCatalogApiClient.php

```diff
@@ -69,4 +69,37 @@ private function getPimUrl(): string
 
         return $pimUrl;
     }
+
+    /**
+     * @return array<mixed>
+     */
+    public function getCatalogProducts(string $catalogId, int $limit = 10, string $searchAfter = null, string $updatedAfter = null, string $updatedBefore = null): array
```

Comment:
Why don't you add the `?` on type-hinting?
```suggestion
    public function getCatalogProducts(string $catalogId, int $limit = 10, ?string $searchAfter = null, ?string $updatedAfter = null, ?string $updatedBefore = null): array
```